src/proof/theory_proof.cpp

```diff
@@ -879,17 +896,28 @@ void LFSCTheoryProofEngine::printCoreTerm(Expr term, std::ostream& os, const Pro
 
   switch(k) {
   case kind::ITE: {
-    os << (term.getType().isBoolean() ? "(ifte ": "(ite _ ");
-
-    bool booleanCase = term[0].getType().isBoolean();
-    if (booleanCase && printsAsBool(term[0])) os << "(p_app ";
-    printBoundTerm(term[0], os, map);
-    if (booleanCase && printsAsBool(term[0])) os << ")";
+    bool usePredicateType = term.getType().isBoolean();
+    os << (usePredicateType ? "(ifte " : "(ite _ ");
 
+    printBoundFormula(term[0], os, map);
     os << " ";
-    printBoundTerm(term[1], os, map);
+    if (usePredicateType)
```

Comment:
Shouldn't this be updated to `term[1].getType().isBoolean()` in case the results of the ITE are booleans?

Reply:
I believe that our typing rules dictate that that the arms of an ITE have the same type as the ITE expression itself. Thus, it suffices to check the type of the ITE itself.

I'll add an assertion for this though!

Reply:
Revisiting this, I was wrong: the arms can actually be strict subtypes of the ITE.

However, we still want to use `usePredicateType`, redubbed `useFormulaType`, because if we choose to use `ifte`, the arms have to be formulas.